crypto/hd/algo.go

```diff
@@ -69,3 +73,27 @@ func (s secp256k1Algo) Generate() GenerateFn {
 		return &secp256k1.PrivKey{Key: bzArr}
 	}
 }
+
+type sr25519Algo struct {
+}
+
+func (s sr25519Algo) Name() PubKeyType {
+	return Sr25519Type
+}
+
+// Derive derives and returns the sr25519 private key for the given seed and HD path.
+func (s sr25519Algo) Derive() DeriveFn {
+	return deriveSr25519
+}
+
+func deriveSr25519(mnemonic string, bip39Passphrase, hdPath string) ([]byte, error) {
+	seed, err := schnorrkel.SeedFromMnemonic(mnemonic, bip39Passphrase)
+	return seed[:], err
+}
```

Comment:
`hdPath` is unused and [SeedFromMnemonic](https://github.com/ChainSafe/go-schnorrkel/blob/58622d036665650cbfdd86f819c3a56bbc0327dd/bip39.go#L28) is simply a BIP39 ["From mnemonic to seed"](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#from-mnemonic-to-seed) implementation, i.e. a hash if the (mnemonic, password). So `Derive()` is very misleading.

The implementation of `SeedFromMnemonic` is not complient to the standard by the way because unicode normalization of the password is missing.

Reply:
The author of https://github.com/paritytech/substrate-bip39 argues that unicode normalization is not a good idea for mnemonics which contains some truth, especially given the problems that occured with non-English wordlists. However, he seems to have missed that unicode normalization matters for password. At the end of the day an implementation was created that is almost BIP39, but not quite. But it is called substrate-bip39. Congratulations. Hope it works for the Parity bubble but are we really blindly copying this unspecified stuff?